src/Elliptic/DiscontinuousGalerkin/Initialization.hpp

```diff
@@ -97,24 +113,89 @@ struct InitializeFacesAndMortars {
                  domain::Tags::ElementMap<Dim>,
                  domain::Tags::InverseJacobian<Dim, Frame::ElementLogical,
                                                Frame::Inertial>>;
+  template <typename Background = std::nullptr_t>
   void operator()(
-      gsl::not_null<DirectionMap<Dim, Direction<Dim>>*> face_directions,
-      gsl::not_null<DirectionMap<Dim, tnsr::I<DataVector, Dim>>*>
+      const gsl::not_null<DirectionMap<Dim, Direction<Dim>>*> face_directions,
+      const gsl::not_null<DirectionMap<Dim, tnsr::I<DataVector, Dim>>*>
           face_inertial_coords,
-      gsl::not_null<DirectionMap<Dim, tnsr::i<DataVector, Dim>>*> face_normals,
-      gsl::not_null<DirectionMap<Dim, Scalar<DataVector>>*>
+      const gsl::not_null<DirectionMap<Dim, tnsr::i<DataVector, Dim>>*>
+          face_normals,
+      const gsl::not_null<DirectionMap<Dim, Scalar<DataVector>>*>
           face_normal_magnitudes,
-      gsl::not_null<DirectionMap<Dim, tnsr::ij<DataVector, Dim>>*>
+      const gsl::not_null<DirectionMap<Dim, tnsr::ij<DataVector, Dim>>*>
           deriv_unnormalized_face_normals,
-      gsl::not_null<::dg::MortarMap<Dim, Mesh<Dim - 1>>*> mortar_meshes,
-      gsl::not_null<::dg::MortarMap<Dim, ::dg::MortarSize<Dim - 1>>*>
+      const gsl::not_null<::dg::MortarMap<Dim, Mesh<Dim - 1>>*> mortar_meshes,
+      const gsl::not_null<::dg::MortarMap<Dim, ::dg::MortarSize<Dim - 1>>*>
           mortar_sizes,
       const Mesh<Dim>& mesh, const Element<Dim>& element,
       const ElementMap<Dim, Frame::Inertial>& element_map,
       const InverseJacobian<DataVector, Dim, Frame::ElementLogical,
                             Frame::Inertial>& inv_jacobian,
-      const std::vector<std::array<size_t, Dim>>& initial_extents)
-      const;
+      const std::vector<std::array<size_t, Dim>>& initial_extents,
+      const Background& background = std::nullptr_t{}) const {
+    static_assert(std::is_same_v<InvMetricTag, void> or
+                      not(std::is_same_v<Background, std::nullptr_t>),
+                  "Supply an analytic background from which the 'InvMetricTag' "
+                  "can be retrieved");
+    const Spectral::Quadrature quadrature = mesh.quadrature(0);
```

Comment:
Does this assume isotropic quadrature? If so, should there be an `ASSERT` for that?

Reply:
yep, done

Reply:
squashed in this snipped:

```cpp
    ASSERT(std::equal(mesh.quadrature().begin() + 1, mesh.quadrature().end(),
                      mesh.quadrature().begin()),
           "This function is implemented assuming the quadrature is isotropic");
```